modules/precisoBidAdapter.js

```diff
@@ -23,7 +23,7 @@ export const spec = {
   buildRequests: (validBidRequests = [], bidderRequest) => {
     // convert Native ORTB definition to old-style prebid native definition
     validBidRequests = convertOrtbRequestToProprietaryNative(validBidRequests);
-    // userId = validBidRequests[0].userId.pubcid;
+    userId = validBidRequests[0].userId.pubcid;
```

Comment:
It is possible that the `userId` object is `undefined` / `null` which would cause this to error and not make a request to your exchange.

You can update this to

`userId = validBidRequests[0].userId?.pubcid;`

Note, if `userId.pubcid` does not exist, then your `userId` var will be set to `undefined` and sent as empty string in your userSync URL like so

`${URL_SYNC}id=&gdpr`;